seb_openedx/middleware.py

```diff
@@ -38,24 +39,34 @@ def process_view(self, request, view_func, view_args, view_kwargs):
                 # Second: Granular white-listing
                 access_denied = True
 
+            banned = False
+            user_name = request.user.username if hasattr(request, 'user') else None
+
             active_comps = get_enabled_permission_classes()
             for permission in active_comps:
                 if permission().check(request, course_key):
                     access_denied = False
 
-        if access_denied:
-            return self.handle_access_denied(request, view_func, view_args, view_kwargs, course_key)
+            if BANNING_ENABLED and user_name and is_user_banned(user_name, course_key):
+                banned = True
+                access_denied = True
+
+            if access_denied:
+                return self.handle_access_denied(request, view_func, view_args, view_kwargs, course_key, user_name, banned)
 
         return None
 
     # pylint: disable=too-many-arguments
-    def handle_access_denied(self, request, view_func, view_args, view_kwargs, course_key):
+    def handle_access_denied(self, request, view_func, view_args, view_kwargs, course_key, user_name, banned):
```

Comment:
This does look like we are starting to put too many arguments. Do we really want to handle banning inside of this function?

Reply:
The view needs to know about the banning; how else than as an argument?

Reply:
We can make a function called handle_banning, which is called on line 51 and use a property, so at any point the methods can call `self.is_banned` o `self.access_denied`